### PR TITLE
Sprint1

### PR DESCRIPTION
**Pull request by GregoryForrest**
Remove accidental TMCR carryover
Breaking changes to Combobox to provide a better UI
Add CAFTOPInfo page with fields:
 - Program Group
 - Program Name
 - Program Element Code (PEC)
 - Lead Command
 - Preparing Base
 - Preparing Office
 - Program Manager(s)
    - First Name
    - Last Name
    - DSN
    - Email

### CODE REVIEW

File: src/Steps/Info/Fields/ProgramElementCode.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    <BACDropdown<CAFTOPInfo>
+      name="ProgramElementCode"
+      labelText="Program Element Code (PEC)"
+      rules={{ required: true }}
+    >
+      {availablePECS.map((item) => (
+        <Option key={item} value={item} text={item}>
+          {item}
+        </Option>
+      ))}
```
Comment (RobertPorterfield) [inline src/Steps/Info/Fields/ProgramElementCode.tsx:32]: Suggestion: if `availablePECS.length === 1` go ahead and pre-select the one item

File: src/Steps/Info/Info.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      dispatch({ type: "PREV_STEP" });
+    }
+    return Promise.resolve();
+  };
+
+  const submitError: SubmitErrorHandler<Partial<CAFTOPInfo>> = (_data) => {
+    console.log(_data);
+    window.alert(
+      "Please correct errors on this page before navigating to another page"
+    );
```
Comment (RobertPorterfield) [inline src/Steps/Info/Info.tsx]: Not entirely sure this alert is needed. (I also strongly dislike window.alert, so I could be biased). I think the scroll to first error with the red validation text is probably enough.

File: src/Steps/Info/Fields/PreparingOffice.tsx
Comment (RobertPorterfield) [inline src/Steps/Info/Fields/PreparingOffice.tsx:1]: Is it possible to trigger validation on this field prior to hitting submit? My thinking is that the moment I hit 21 characters I'd like to see the validation error. Or, the moment I "leave" the field I'd get the validation error

File: src/Steps/Info/Fields/ProgramElementCode.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+export const ProgramElementCodeRuleFinal = z.object({
+  ProgramElementCode: finalRule,
+});
+
+export const ProgramElementCode = () => {
+  const ProgramNames = useProgramNamesAndECs();
+  const programName = useWatch<CAFTOPInfo>({ name: "ProgramName" });
+  const availablePECS =
+    ProgramNames?.data?.find((item) => item.Title === programName)?.PECs ?? [];
```
Comment (RobertPorterfield) [inline src/Steps/Info/Fields/ProgramElementCode.tsx:21]: I think you need some additional logic that clears the selected PEC when Program Name changes

File: src/Steps/Info/Fields/PreparingOffice.tsx
Comment (GregoryForrest) [reply src/Steps/Info/Fields/PreparingOffice.tsx:1]: Great suggestion!  Changed the form mode to be "onChange" like we have with other forms, so they get instant feedback.

File: src/Steps/Info/Info.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    <>
+      <Title1>CAFTOP Information Page</Title1>
+      <Title2>General CAFTOP Selections...</Title2>
+
+      <FormProvider {...myForm}>
+        <form
+          id="innerForm"
+          onSubmit={(...args) =>
+            myForm.handleSubmit(submitSuccess, submitError)(...args)
+          }
```
Comment (RobertPorterfield) [inline src/Steps/Info/Info.tsx]: What's the purpose of the extra `(...args)` function call here? Does this do something I'm not seeing? Should 63-65 not just be
`onSubmit={myForm.handleSubmit(submitSuccess, submitError)}`

File: src/Steps/Info/Fields/ProgramElementCode.tsx (hunk 3 above)
Comment (GregoryForrest) [reply src/Steps/Info/Fields/ProgramElementCode.tsx:21]: Great UI improvement suggestion.  Incorporated it with setting the PEC if there is only 1 available.

File: src/Steps/Info/Fields/ProgramElementCode.tsx (hunk 1 above)
Comment (GregoryForrest) [reply src/Steps/Info/Fields/ProgramElementCode.tsx:32]: Great UI improvement suggestion.  Incorporated it with clearing the PEC on Program Name clearing/changing.

File: src/Steps/Info/Info.tsx (hunk 2 above)
Comment (GregoryForrest) [reply src/Steps/Info/Info.tsx]: Completely agree.  Removed the unneeded ugly browser alert.

File: src/utilities/Validations.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+import { ProgramGroupRuleFinal } from "Steps/Info/Fields/ProgramGroup";
+import { ProgramManagersRuleFinal } from "Steps/Info/Fields/ProgramManagers";
+import { ProgramNameRuleFinal } from "Steps/Info/Fields/ProgramName";
+import { TechOrderManagerRuleFinal } from "Steps/Info/Fields/TechOrderManager";
+import { useProgramNamesAndECs } from "api/ProgramNamesAndElementCodes";
+import { ZodSchema, z } from "zod";
+
+const useAddlPECValidation = (schema: ZodSchema) => {
+  const ProgramNamesAndECs = useProgramNamesAndECs();
+  return schema.superRefine((data, ctx) => {
```
Comment (RobertPorterfield) [inline src/utilities/Validations.ts:15]: I think you can type your data to `(data: CAFTOPInfo, ctx)`

File: src/Steps/Info/Info.tsx (hunk 4 above)
Comment (GregoryForrest) [reply src/Steps/Info/Info.tsx]: Great question.  It's a workaround to remove a lint warning for "Promise-returning function provided to attribute where a void return was expected  @typescript-eslint/no-misused-promises".  However I did forget to include one part "void".

GH entry I was using for the workaround: https://github.com/orgs/react-hook-form/discussions/8020

File: src/utilities/Validations.ts (hunk 5 above)
Comment (GregoryForrest) [reply src/utilities/Validations.ts:15]: Good call!  Eliminated 3 lint warnings!

File: src/Steps/Info/Info.tsx (hunk 4 above)
Comment (RobertPorterfield) [reply src/Steps/Info/Info.tsx]: Ooph. I see.

**Review by RobertPorterfield — APPROVED** [review]
Excellent work! Very clean. 🤘

**Pull request merged**